### DISCUSSION TITLE
Unstable Web Search Functionnality

### DISCUSSION THREAD

**Discussion by Genai-labs** [open-webui/open-webui#7525] [Issues]
# Bug Report

## Environment

- **Open WebUI Version:** [v.0.4.6]

## Expected Behavior:

The web search functionality should scrape sufficient data from retrieved URLs to generate complete and accurate responses for queries

## Actual Behavior:

When the scraping process fails to collect enough information from certain URLs (e.g., only the title is retrieved), the model either produces incomplete responses or triggers errors such as: "I do not have the ability to search for real-time data"

## Description

**Bug Summary:**

The web search feature sometimes encounters issues when scraping data from retrieved URLs. For some sites, only limited information (e.g., the title) is extracted, which prevents the model from forming a comprehensive response. This leads to incomplete outputs or error messages indicating an inability to perform real-time searches.

## Reproduction Details

**Steps to Reproduce:**

1 - Perform web searches that rely on data from a variety of websites.
2 - Observe cases where the retrieved information is limited, leading to incomplete responses or errors.

## Additional Information

- This issue is independent of the search engine used (tested with DuckDuckGo and Bing).

## Potential Enhancement:

It may be worth considering the exclusion of URLs that do not contain sufficient or usable scraped text during the response generation process. By filtering out these URLs, the system could focus only on sources that provide meaningful data, potentially improving the quality and completeness of the generated response

**Comment by tjbck** [comment]
Keep us updated on your troubleshooting journey!

**Comment by spammenotinoz** [comment]
It helps if you update your system prompt so the LLM knows it has access to the internet, otherwise the LLM does not know.

**Comment by unicorn667** [comment]
do you use a docker-compose file and postgresql ?

**Comment by mikygit** [comment]
Same problem here: https://github.com/open-webui/open-webui/discussions/7725
I'm using searXNG so i don't think it's only a matter of not having enough content from the web search actually.
When i check the logs sometimes when it 'fails', it appears the llm have enough information to answer properly.
Also, i'm not sure how the previous answers cache does not impact the llm response ...
